quixstreams/kafka/configuration.py

```diff
@@ -0,0 +1,147 @@
+from typing import Literal, Optional, Callable, Tuple, List, get_args
+
+from pydantic import AliasGenerator, SecretStr, AliasChoices, Field
+from pydantic.functional_validators import BeforeValidator
+from pydantic_settings import BaseSettings, SettingsConfigDict
+from typing_extensions import Self, Annotated
+
+__all__ = ("ConnectionConfig",)
+
+MECHANISM_ALIAS = AliasChoices("sasl_mechanism", "sasl_mechanisms")
+
+
+class ConnectionConfig(BaseSettings):
```

Comment:
There's one extra thing I'd add to the class.
By default, `BaseSettings` will also pull values from the environment, which is not this class's primary goal.
The parameters are supposed to be passed directly.

To turn off this behavior, we need to add an extra method:
```
@classmethod
    def settings_customise_sources(
        cls,
        settings_cls: Type[BaseSettings],
        init_settings: PydanticBaseSettingsSource,
        env_settings: PydanticBaseSettingsSource,
        dotenv_settings: PydanticBaseSettingsSource,
        file_secret_settings: PydanticBaseSettingsSource,
    ) -> Tuple[PydanticBaseSettingsSource, ...]:
        return (init_settings,)
```

It will make the `BaseSettings` class ignore every other source of settings except `__init__`.
Otherwise, there may be clashes if the env variable with the same name is defined